### PR TITLE
Adding devcontainer for GitHub remote-extensions and codespaces

### PR DESCRIPTION
**Pull request by sw-yx**
## What was changed
Adding devcontainer for GitHub remote-extensions and codespaces

image

## Why?
Run any sample in one click

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+// Update the VARIANT arg in docker-compose.yml to pick a Node.js version
+{
+	"name": "Samples Typescript",
+	"dockerComposeFile": "docker-compose.yml",
+	"service": "samples",
+	"workspaceFolder": "/workspace",
+	"shutdownAction": "stopCompose",
+	// Set *default* container specific settings.json values on container create.
+	"settings": {
+		"sqltools.connections": [
```
Comment (jbreiding) [inline .devcontainer/devcontainer.json:12]: This should be removed since temporalite is being used now instead of postgres

**Review by vkarpov15 — APPROVED** [review]
Went through this and got it working on my machine:

image

I second the suggestion to remove postgres - my biggest concern is that setting up the remote extension was very slow, took at least 30 mins to build. The more we can remove the better :)

**Comment by vkarpov15** [comment]
Also, please add `.devcontainer` to `.prettierignore`

**Comment by jbreiding** [comment]
> Went through this and got it working on my machine:
> 
> image
> 
> I second the suggestion to remove postgres - my biggest concern is that setting up the remote extension was very slow, took at least 30 mins to build. The more we can remove the better :)

When you say "my machine", what are you using the get/maintain a docker context?

If you have a VM backing this, and using default docker-desktop settings, the performance would be pretty poor.

**Comment by vkarpov15** [comment]
I'm not sure how to answer "what are you using to get/maintain a docker context?", I'm afraid my understanding of Docker is pretty limited. I followed the instructions here: https://code.visualstudio.com/docs/remote/containers

**Comment by sw-yx** [comment]
k gonna merge it in now since i cant edit the thing otherwise

**Pull request merged**